clients/js/src/embeddings/AmazonBedrockEmbeddingFunction.ts

```diff
@@ -0,0 +1,74 @@
+import { IEmbeddingFunction } from "./IEmbeddingFunction";
+
+let amazonBedrockApi: any;
+
+export class AmazonBedrockEmbeddingFunction implements IEmbeddingFunction {
+  private model: string;
+  private configuration: any; // Assume this is BedrockRuntimeClientConfig
+  private amazonBedrockApi?: any;
+  private invokeCommand?: any;
+
+  constructor({ config, model }: { config: any; model?: string}) {
+    this.configuration = config;
+    if (!this.configuration.region) {
+      this.configuration.region = "us-east-1";
+    }
+    this.model = model || "amazon.titan-embed-text-v1";
+  }
+
+  private async loadClient() {
+    if (this.amazonBedrockApi) return;
+    try {
+      // eslint-disable-next-line global-require,import/no-extraneous-dependencies
+      const { amazonBedrock } = await AmazonBedrockEmbeddingFunction.import();
+      amazonBedrockApi = new amazonBedrock.BedrockRuntimeClient(
+        this.configuration
+      );
+      this.invokeCommand = amazonBedrock.InvokeModelCommand;
+    } catch (_a) {
+      // @ts-ignore
+      if (_a.code === "MODULE_NOT_FOUND") {
+        throw new Error(
+          "Please install the @aws-sdk/client-bedrock-runtime package to use the AmazonBedrockEmbeddingFunction, `npm install -S @aws-sdk/client-bedrock-runtime`"
+        );
+      }
+      throw _a; // Re-throw other errors
+    }
+    this.amazonBedrockApi = amazonBedrockApi;
+  }
+
+  public async generate(texts: string[]): Promise<number[][]> {
+    await this.loadClient();
+    const td = new TextDecoder();
+    const embeddings: number[][] = [];
+    return Promise.all(texts.map(async text => {
```

Comment:
@chezou, just confirming my understanding here is that if the user provides several texts, the EF will embed each one individually, and if it hits a problem, then an error is raised. The existing embeddings are, in a way, lost, correct?

Reply:
Yes, that's the correct understanding. If we need to preserve partial embeddings, I have to modify to use `Promise.allSettled` instead.

Reply:
@tazarov Let me know if I need to do something additional.